build/release/githubRelease.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"bosun.org/_third_party/github.com/google/go-github/github"
+)
+
+type myRoundTripper struct {
+	accessToken string
+}
+
+func (rt myRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("Authorization", fmt.Sprintf("token %s", rt.accessToken))
+	return http.DefaultTransport.RoundTrip(r)
+}
+
+var (
+	client *github.Client
+	number string
+	binDir string
+	sha    string
+)
+
+func init() {
+	accessToken := os.Getenv("GITHUB_ACCESS_TOKEN")
+	if accessToken == "" {
+		log.Fatal("GITHUB_ACCESS_TOKEN env required")
+	}
+	if number = os.Getenv("BUILD_NUMBER"); number == "" {
+		log.Fatal("BUILD_NUMBER env required")
+	}
+	if sha = os.Getenv("GITHUB_VERSION_SHA"); sha == "" {
+		log.Fatal("GITHUB_VERSION_SHA env required")
+	}
+	binDir = os.Getenv("OUTPUTDIR")
+	hClient := &http.Client{Transport: myRoundTripper{accessToken}}
+	client = github.NewClient(hClient)
+}
+
+func main() {
+	fmt.Printf("Releasing to github %s.\n", number)
+
+	fmt.Println("Finding last release TODO: use latest endpoint once merged into go-github...")
+	releases, _, err := client.Repositories.ListReleases("bosun-monitor", "bosun", &github.ListOptions{PerPage: 100})
+	if err != nil {
+		log.Fatal(err)
+	}
+	var lastRelease github.RepositoryRelease
+	for _, r := range releases {
+		if r.PublishedAt != nil {
+			lastRelease = r
+			break
+		}
+	}
+
+	fmt.Println("Getting all prs since last release to build release notes...")
```

Comment:
The idea is that pr titles are more likely to be feature descriptions than commit messages